Optionally listen to native VK_PASTE key

SquidDev opened the pull request:
Also ensure the repeat parameter of key down events are fired. Closes #38.

This doesn't appear to work on Windows, so I'd appreciate testing on other platforms. If it doesn't work there then I'll cherry pick the other fixes directly.

I'm sorry about the whitespace changes, but I didn't really want to go and fix each one manually. Can everyone make sure whitespace is trimmed in their editor configuration?

SquidDev committed:
Optionally listen to native VK_PASTE key

Also ensure the repeat parameter of key down events are fired.

Lignum reviewed (changes requested):
This doesn't seem to work correctly at all on Linux. Non-native paste is completely broken, doing nothing at all, and native paste only allows Ctrl+V, no middle click, as would be expected (is the config setting inverted by accident?).

Additionally, key repeat no longer works, making deleting things by holding backspace impossible.

SquidDev commented:
> is the config setting inverted by accident?

Yes. I'm a plonker. Inversion should be fixed, though I doubt native presses are. I've added a comment to the documentation saying as much.

> Additionally, key repeat no longer works, making deleting things by holding backspace impossible.

Did I mention I'm a plonker? Also fixed.

---

The alternative here would be to offer various "paste modes", configurable from an enum or something. Off the top of my head, I can think of:

 - Ctrl+V
 - Middle click
 - Shift + Insert

Lignum commented:
> Inversion should be fixed, though I doubt native presses are.

Can confirm that inversion's fixed, and indeed native presses still don't do anything in any mode.

> The alternative here would be to offer various "paste modes", configurable from an enum or something. Off the top of my head, I can think of:
> * Ctrl+V
> * Middle click
> * Shift + Insert

Hm, that seems a bit excessive, let's add it if it seems to be something people need.

Either way, if `VK_PASTE` isn't firing, that's not our fault, that's Java's mess. I got the `VK_PASTE` idea from [this StackOverflow question](https://stackoverflow.com/questions/9677855/java-listen-for-system-safe-paste-command), where people are also commenting that it doesn't work. However, the top answer suggests to use a `DocumentListener`, I'm not sure if we can use that here, but if so, then that might work.

SquidDev committed:
Attempt to use the current look and feel's paste events instead.
